wordpress-mobile/WordPress-Android
# Issue/4764 self hosted block post error

Fixes #4764 

Provides a better description for each error and allows to access the Helpshift help pages from a button in SignInDialogFragment.

To test:
1. sign in to a self hosted site that blocks POST requests (so XML-RPC sign in won't work)
2. get the error screen and check the more specific error.
3. Also, now you can tap on "Tell me more" and it will lead you to a more specific description of the problem in our Helpshift FAQ page or section

Check video here https://cloudup.com/cK3DbBPIyRa

cc @maxme @tonyr59h @aforcier 

Note: please check `build.gradle` file as it points to the branch @aforcier was working on
